### ISSUE TITLE
kops upgrade is required

### ISSUE THREAD

**Issue by barcherf5** [eddycharly/terraform-provider-kops#337]
When I attempt to run terraform apply, it returns an error that is somehow related to kops_cluster_updater:

│ Error: kops upgrade is required
│ 
│   with module.kops.kops_cluster_updater.updater,
│   on .terraform/modules/kops/infrastructure/cluster/kops/updater.tf line 1, in resource "kops_cluster_updater" "updater":
│    1: resource "kops_cluster_updater" "updater" {

Contents of that resource/file:

`  resource "kops_cluster_updater" "updater" {
    cluster_name = kops_cluster.cluster.name
  
    keepers = {
      cluster = kops_cluster.cluster.revision,
      master0 = kops_instance_group.master0.revision
      master1 = kops_instance_group.master1.revision
      master2 = kops_instance_group.master2.revision
      node0 = kops_instance_group.node0.revision
      node1 = kops_instance_group.node1.revision
      node2 = kops_instance_group.node2.revision
    }
  
    apply {
      skip = false
      allow_kops_downgrade = true
    }
  
    rolling_update {
      skip                = false
      fail_on_drain_error = true
      fail_on_validate    = true
      validate_count      = 1
    }
  
    validate {
      skip = false
    }
  
   # ensures rolling update happens after the cluster and instance groups are up to date
    depends_on   = [
      kops_cluster.cluster,
      kops_instance_group.master0,
      kops_instance_group.master1,
      kops_instance_group.master2,
      kops_instance_group.node0,
      kops_instance_group.node1,
      kops_instance_group.node2
    ]
  }
`

My deepest apologies if I have made a dumb mistake. :)

**Comment by barcherf5** [comment]
I don't know why it refuses to format the code properly. Sorry!

**Comment by barcherf5** [comment]
Also, for the record, it doesn't create any EC2 instances when I `terraform apply`
Not sure if that's related.

**Comment by eddycharly** [comment]
It looks like the provider refuses to apply/rollout the cluster.
Is it a brand new cluster you created from scratch ?
What version are you using ?

**Comment by barcherf5** [comment]
Yes, using TF+kOps to create the cluster from scratch. I'm using the latest version.

**Comment by eddycharly** [comment]
Do you have a `kops-version.txt` file in your bucket ?

**Comment by barcherf5** [comment]
No, that file does not exist in the bucket. :(

**Comment by eddycharly** [comment]
It looks strange, this error message usually means you're trying to update a cluster with an old kOps version.
Not sure how i can help on this.

**Comment by clayrisser** [comment]
Another situation that can cause this error is when you use a kubernetes_version that kops does not support. Make sure the kubernetes_version matches one of the versions from the link below.

https://github.com/kubernetes/kops/tags
